jsonrpc/filter_manager_test.go

```diff
@@ -11,6 +13,123 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func Test_GetLogsForQuery(t *testing.T) {
+	t.Parallel()
+
+	blockHash := types.StringToHash("1")
+
+	// Topics we're searching for
+	topic1 := types.StringToHash("4")
+	topic2 := types.StringToHash("5")
+	topic3 := types.StringToHash("6")
+
+	var topics = [][]types.Hash{{topic1}, {topic2}, {topic3}}
+
+	testTable := []struct {
+		name           string
+		query          *LogQuery
+		shouldFail     bool
+		expectedLength int
+	}{
+		{"Found matching logs, fromBlock < toBlock",
+			&LogQuery{
+				fromBlock: 1,
+				toBlock:   3,
+				Topics:    topics,
+			},
+			false, 3},
+		{"Found matching logs, fromBlock == toBlock",
+			&LogQuery{
+				fromBlock: 2,
+				toBlock:   2,
+				Topics:    topics,
+			},
+			false, 1},
+		{"Found matching logs, BlockHash present",
+			&LogQuery{
+				BlockHash: &blockHash,
+				Topics:    topics,
+			},
+			false, 1},
+		{"No logs found", &LogQuery{
+			fromBlock: 4,
+			toBlock:   5,
+			Topics:    topics,
+		}, false, 0},
+		{"Invalid block range", &LogQuery{
+			fromBlock: 10,
+			toBlock:   5,
+			Topics:    topics,
+		}, true, 0},
```

Comment:
Why not check for the exact error here?

Reply:
From what I can see in the code, this test was not designed to receive an error message. 
Fixed 99ca16124f3c6827c1587da7427bda6eae93bb29

Reply:
This should be `assert.ErrorIs` instead of equal

Reply:
Fixed 530d473ab4255e17ba5c8bfe4ba2002787ad1445